File: src/com/prey/actions/HttpDataService.java
/*******************************************************************************
 * Created by Carlos Yaconi
 * Copyright 2012 Fork Ltd. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions;

 
import java.io.InputStream;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

 

public class HttpDataService {

	private String key;
	private HashMap<String, String> dataList;
	private String singleData;
	private boolean isList;
	private String httpMethod;
	private String url;
	private List<InputStream> files;
 

	public HttpDataService(String key) {
		this.key = key;
		dataList = new HashMap<String, String>();
		files= new ArrayList<InputStream>();
	}

	public HashMap<String, String> getDataAsParameters() {
		HashMap<String, String> parameters = new HashMap<String, String>();
		if (isList()) {
			StringBuffer key = new StringBuffer();
			for (String valueKey : dataList.keySet()) {
				String valueData = dataList.get(valueKey);
				key.append(this.key);
				key.append("[");
				key.append(valueKey);
				key.append("]");
				parameters.put(key.toString(), valueData);
				key.delete(0, key.length());
			}
			key = null;
		} else
			parameters.put(key, singleData);

		return parameters;
	}

	public String getDataAsString() {
		StringBuffer sb = new StringBuffer();

		if (isList()) {
			for (String valueKey : dataList.keySet()) {
				String valueData = dataList.get(key);
				sb.append(key);
				sb.append("[");
				sb.append(valueKey);
				sb.append("]=");
				sb.append(valueData);
				sb.append("&");
			}

		} else
			sb.append(key).append("=").append(singleData).append("&");
		return sb.toString();
	}

	public String getKey() {
		return key;
	}

	public void setKey(String key) {
		this.key = key;
	}

	public HashMap<String, String> getDataList() {
		return dataList;
	}

	public void setDataList(HashMap<String, String> dataList) {
		this.dataList = dataList;
	}

	public String getSingleData() {
		return singleData;
	}

	public void setSingleData(String singleData) {
		this.singleData = singleData;
	}

	public boolean isList() {
		return isList;
	}

	public void setList(boolean isList) {
		this.isList = isList;
	}

	public String getHttpMethod() {
		return httpMethod;
	}

	public void setHttpMethod(String httpMethod) {
		this.httpMethod = httpMethod;
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
	}

	public List<InputStream> getFiles() {
		return files;
	}

	public void setFiles(List<InputStream> files) {
		this.files = files;
	}

	public void addFile(InputStream file) {
		files.add(file);
	}


 
 

}

File: app/src/main/java/com/prey/actions/kotlin/HttpDataService.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.kotlin

import com.prey.net.http.kotlin.EntityFile

class HttpDataService(private var keyValue: String) {
    private var dataList: MutableMap<String, String?>? = HashMap()
    private var singleData: String? = null
    private var isList: Boolean = false
    private var httpMethod: String? = null
    private var url: String? = null
    private var entityFiles: List<EntityFile>

    fun getKeyValue(): String {
        return keyValue
    }

    fun setKeyValue(keyValue: String?) {
        this.keyValue = keyValue!!
    }

    fun getDataList(): MutableMap<String, String?>? {
        return dataList
    }

    fun setDataList(dataList: MutableMap<String, String?>?) {
        this.dataList = dataList!!
    }

    fun isList(): Boolean {
        return isList
    }

    fun setList(isList: Boolean) {
        this.isList = isList
    }


    fun getSingleData(): String {
        return singleData!!
    }

    fun setSingleData(singleData: String?) {
        this.singleData = singleData
    }

    fun addDataListAll(map: MutableMap<String, String?>?) {
        dataList!!.putAll(map!!)
    }

    init {
        entityFiles = ArrayList()
    }

    fun getReportAsParameters(): MutableMap<String, String?> {
        val parameters = HashMap<String, String?>()
        if (isList) {
            var key: StringBuffer? = StringBuffer()
            for (valueKey in dataList!!.keys) {
                val valueData = dataList!![valueKey]
                key!!.append("")
                key.append(this.getKeyValue())
                key.append("[")
                key.append(valueKey)
                key.append("]")
                parameters[key.toString()] = valueData
                key.delete(0, key.length)
            }
            key = null
        } else parameters[" $keyValue "] = singleData
        return parameters
    }

    fun getDataAsParameters(): MutableMap<String, String?> {

        val parameters = HashMap<String, String?>()
        if (isList) {
            var key: StringBuffer? = StringBuffer()
            for (valueKey in dataList!!.keys) {
                val valueData = dataList!![valueKey]
                key!!.append(this.keyValue)
                key.append("[")
                key.append(valueKey)
                key.append("]")
                parameters[key.toString()] = valueData
                key.delete(0, key.length)
            }
            key = null
        } else parameters[keyValue] = singleData
        return parameters
    }

    fun getDataAsString(): String {
        val sb = StringBuffer()
        if (isList) {
            for (valueKey in dataList!!.keys) {
                val valueData = dataList!![keyValue]
                sb.append(keyValue)
                sb.append("[")
                sb.append(valueKey)
                sb.append("]=")
                sb.append(valueData)
                sb.append("&")
            }
        } else sb.append(keyValue).append("=").append(singleData).append("&")
        return sb.toString()
    }

    fun getDataListKey(key: String): String? {
        var value: String? = ""
        if (dataList != null && dataList!!.containsKey(key)) {
            value = dataList!![key]
        }
        return value
    }

    fun putData(dataList: Map<String, String>?) {
        this.dataList!!.putAll(dataList!!)
    }

    fun addEntityFile(entityFile: EntityFile) {
        entityFiles.plus(entityFile)
    }

    fun getEntityFiles(): List<EntityFile> {
        return entityFiles
    }

    fun setEntityFiles(entityFiles: List<EntityFile>) {
        this.entityFiles = entityFiles
    }
}